nevergrad/functions/photonics/core.py

```diff
@@ -115,22 +125,32 @@ class Photonics(base.ExperimentFunction):
       Moosh: A Numerical Swiss Army Knife for the Optics of Multilayers in Octave/Matlab. Journal of Open Research Software, 4(1), p.e13.
     """
 
-    def __init__(self, name: str, dimension: int, bounding_method: str = "clipping", rolling: bool = False) -> None:
+    def __init__(self, name: str, dimension: int, bounding_method: str = "clipping", rolling: bool = False, as_tuple=False) -> None:
         assert name in ["bragg", "morpho", "chirped"]
         self.name = name
+        self._as_tuple = as_tuple
         self._base_func = {"morpho": photonics.morpho, "bragg": photonics.bragg, "chirped": photonics.chirped}[name]
-        param = _make_parametrization(name=name, dimension=dimension, bounding_method=bounding_method, rolling=rolling)
+        param = _make_parametrization(name=name, dimension=dimension,
+            bounding_method=bounding_method, rolling=rolling, as_tuple=as_tuple)
         super().__init__(self._compute, param)
 
     # pylint: disable=arguments-differ
-    def evaluation_function(self, x: np.ndarray) -> float:  # type: ignore
+    def evaluation_function(self,  *args: tp.Any, **kwargs: tp.Any) -> float:  # type: ignore
+        assert len(kwargs.items()) == 0
+        x = np.concatenate([a for a in args])
+        if self._as_tuple:
+            x = np.transpose(x)
```

Comment:
x
```suggestion
        x = args[0]
        if self._as_tuple:
            x = np.transpose(np.concatenate([a for a in args]))
```
Same below, but factorizing would be even better